crates/sui-graphql-rpc/src/test_infra/cluster.rs

```diff
@@ -230,26 +230,43 @@ pub async fn start_test_indexer_v2(
 }
 
 impl ExecutorCluster {
-    pub async fn wait_for_checkpoint_catchup(&self, checkpoint: u64, timeout: Duration) {
-        async fn inner(s: &ExecutorCluster, checkpoint: u64) {
-            let mut highest_checkpoint = s
+    pub async fn wait_for_checkpoint_catchup(&self, checkpoint: u64, base_timeout: Duration) {
```

Comment:
This change seems more complicated than it needs to be for what it's trying to achieve.  It also includes a bug where it will immediately panic if the checkpoint you are asking for has already been reached, because it sets a timeout of `0`.

What about this:

```rust
pub async fn wait_for_checkpoint_catchup(&self, checkpoint: u64, base_timeout: Duration) {
    let current_checkpoint = s
        .indexer_store
        .get_latest_tx_checkpoint_sequence_number();
        .await
        .unwrap()
        .unwrap();

    let checkpoint_diff = std::min(1, checkpoint.saturating_sub(current_checkpoint));
    let timeout = base_timeout.mul_f64(checkpoint_diff as f64);

    // Everything else as before...
}
```

Reply:
i will rebase off a separate PR to fix this